src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,41 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(health, name) {
+    this.health = 100;
+    this.name = name;
+    Animal.alive.push(this);
+  }
+  static removeFromAlive(animal) {
+    Animal.alive = Animal.alive.filter((a) => a !== animal);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(health, name, hidden) {
+    super(health, name);
+    this.hidden = false;
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (
+      animal instanceof Carnivore ||
+      (animal.hidden && animal instanceof Herbivore)
+    ) {
+      return;
+    }
+    animal.health -= 50;
```

Comment:
Additionally, the bite handling is a Herbivore behavior and therefore, it would be considered a best practice to implement it inside the `Herbivore` class.
As an **optional** task, could you think of a way to move this logic into the Herbivore class?

Reply:
I don't think I understand, Edu! From what I understand from the exercise, the bite is a carnivore method: "Carnivore has a `bite` method, which takes a herbivore object and decreases the object's health by `50`. The method does not work if it is another сarnivore, or the herbivore is currently hiding.". That's why I put it in the class of carnivores, not herbivores. The bite is from carnivore, not herbivore.